### ISSUE TITLE
Some comments on Advanced Forwarding configFile functionality

### ISSUE THREAD

**Issue by liang-hiwin** [TechnitiumSoftware/DnsServer#571]
By looking at [](https://github.com/TechnitiumSoftware/DnsServer/issues/489), we know that dnsserver supports the function of importing a large number of forwarding domain names. The landlord @Potterli20 provided adghome file is too large and not conducive to modification, you can do it in the next version improve. For example, support such writing;

**Comment by liang-hiwin** [comment]
The domain name matching rules for the domains.txt file follow the following rules:

```
for example:
domain:google.com                     #####domian: can be omitted, it will match its own google.com, and its subdomain www.google.com, maps.l.google.com,

full:google.com                           ### will only match itself
keyword:google.com                  ### will match the domain name containing this field, such as google.com.hk, www.google.com.hk
regexp:.+\.google\.com$            ###regular match
```

**Comment by ShreyasZare** [comment]
Thanks for the post. Yes, the app needs some better way to have bulk domain list similar to adguard format. Will get something like you say added to it.

**Comment by liang-hiwin** [comment]
> Thanks for the post. Yes, the app needs some better way to have bulk domain list similar to adguard format. Will get something like you say added to it.

Yes, the current scheme is not convenient to modify

**Comment by liang-hiwin** [comment]
```
          "domains": [
            "example.com"
          ]

```
I have a question, can the domain name "example.com" of the above code match "www.example.com", "example.com"?

**Comment by ShreyasZare** [comment]
> ```
>           "domains": [
>             "example.com"
>           ]
> ```
> 
> I have a question, can the domain name "example.com" of the above code match "www.example.com", "example.com"?

Yes, the domain name and all of its subdomain names will be forwarded.

**Comment by liang-hiwin** [comment]
> > ```
> >           "domains": [
> >             "example.com"
> >           ]
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > I have a question, can the domain name "example.com" of the above code match "www.example.com", "example.com"?
> 
> Yes, the domain name and all of its subdomain names will be forwarded.

I think when I forward a certain domain name request in Advanced Forwarding configuration, the Response Type type in the Query Logs app should be marked as Forward instead of Recursive

**Comment by ShreyasZare** [comment]
> I think when I forward a certain domain name request in Advanced Forwarding configuration, the Response Type type in the Query Logs app should be marked as Forward instead of Recursive,what do you think?
> 
> ![$EZ@UJA{R{2LGX(C7{9}JKB](https://user-images.githubusercontent.com/17548936/221582102-c597f088-2bdd-4c4b-8398-4c6d1567132b.png)

The DNS server considers both recursive resolution and forwarding same for stats and logs. This is even true for DNS server's forwarders settings.

**Comment by ShreyasZare** [comment]
> > Thanks for the post. Yes, the app needs some better way to have bulk domain list similar to adguard format. Will get something like you say added to it.
> 
> It is suggested to press this button first, otherwise my file cannot be enabled. Now my file has nearly reached 300m

If you open the app config and click save then it will attempt to reload the files immediately. But, it checks if the file has modified every minute too so it will auto reload within a minute when the file has changed.
